cluster/repository/cluster_blackbox_test.go

```diff
@@ -37,6 +41,81 @@ func (s *clusterTestSuite) TestCreateAndLoadClusterOK() {
 	test.AssertEqualClusters(s.T(), cluster1, loaded)
 }
 
+func (s *clusterTestSuite) TestCreateAndLoadClusterByURLOK() {
+	cluster1 := test.CreateCluster(s.T(), s.DB)
+	test.CreateCluster(s.T(), s.DB) // noise
+	loaded, err := s.repo.LoadClusterByURL(context.Background(), cluster1.URL)
+	require.NoError(s.T(), err)
+
+	test.AssertEqualClusters(s.T(), cluster1, loaded)
+}
+
+func (s *clusterTestSuite) TestCreateAndLoadClusterByURLFail() {
+	test.CreateCluster(s.T(), s.DB)
+	test.CreateCluster(s.T(), s.DB) // noise
+	loaded, err := s.repo.LoadClusterByURL(context.Background(), uuid.NewV4().String())
+	assert.Nil(s.T(), loaded)
+	test.AssertError(s.T(), err, gorm.ErrRecordNotFound, "record not found")
+}
+
+func (s *clusterTestSuite) TestCreateOKInCreateOrSave() {
+	cluster := test.NewCluster()
+	s.repo.CreateOrSave(context.Background(), cluster)
+	test.CreateCluster(s.T(), s.DB) // noise
+
+	loaded, err := s.repo.LoadClusterByURL(context.Background(), cluster.URL)
+	require.NoError(s.T(), err)
+
+	test.AssertEqualClusters(s.T(), cluster, loaded)
+}
+
+func (s *clusterTestSuite) TestSaveOKInCreateOrSave() {
+	cluster := test.NewCluster()
+	test.CreateCluster(s.T(), s.DB) // noise
+	s.repo.CreateOrSave(context.Background(), cluster)
+
+	loaded, err := s.repo.LoadClusterByURL(context.Background(), cluster.URL)
+	require.NoError(s.T(), err)
+
+	test.AssertEqualClusters(s.T(), cluster, loaded)
+
+	// update cluster details
+	cluster.AppDNS = uuid.NewV4().String()
+	cluster.AuthClientID = uuid.NewV4().String()
+	cluster.AuthClientSecret = uuid.NewV4().String()
+	cluster.AuthDefaultScope = uuid.NewV4().String()
+	cluster.ConsoleURL = uuid.NewV4().String()
+	cluster.LoggingURL = uuid.NewV4().String()
+	cluster.MetricsURL = uuid.NewV4().String()
+	cluster.Name = uuid.NewV4().String()
+	cluster.SaToken = uuid.NewV4().String()
+	cluster.SaUsername = uuid.NewV4().String()
+	cluster.TokenProviderID = uuid.NewV4().String()
+	cluster.Type = uuid.NewV4().String()
+
+	s.repo.CreateOrSave(context.Background(), cluster)
+	loaded, err = s.repo.LoadClusterByURL(context.Background(), cluster.URL)
+	require.NoError(s.T(), err)
+
+	test.AssertEqualClusters(s.T(), cluster, loaded)
+}
+
+func (s *clusterTestSuite) TestCreateOrSaveOSOClusterOK() {
+	clusterConfig, err := configuration.NewConfigurationData("", "./../../configuration/conf-files/oso-clusters.conf")
+	fmt.Println(clusterConfig.GetOSOClusters())
+	require.Nil(s.T(), err)
```

Comment:
require.NoError(s.T(), err)

Reply:
done in 191866f390938551a9cc4caa05f351008db6f508